pure-tests/Network/HTTP/ReqSpec.hs

```diff
@@ -405,6 +409,18 @@ spec = do
         request <- req_ GET url NoReqBody (httpVersion major minor)
         L.requestVersion request `shouldBe` Y.HttpVersion major minor
 
+    describe "quasiquoter" $ do
+      it "works for valid urls"
+        $ property
+        -- Doing it this way instead of just checking if [urlQ|...|] :: (Url
+        -- 'Https, Option _) type checks, so we can catch if the type of scheme
+        -- is unspecified.
+        $ (\(_ :: a, _ :: Option scheme) -> isJust (eqT @a @(Url 'Https)))
```

Comment:
I do not understand what you're trying to achieve here. Can you explain in more details?

Reply:
When implementing I had the problem that `lift` forgot about the type of `scheme`, so `[urlQ|https://example.org/|] :: (Url scheme1, Option scheme2)` instead of `:: (Url Https, Option scheme)`. What I'm doing here is binding the type of the `Url` part to `a` and test using `eqT` if it is `Url 'Https`. If it is `Url Http` that will cause the test to fail, because `eqT @(Url 'Http) @(Url 'Https) = Nothing`, if the type is `Url scheme`, it will fail at compile time complaining that it can't deduce a `Typeable` instance for `Url scheme`.

Reply:
If you remove the generated type signature here the test will fail complaining about a missing `Typeable` instance.

https://github.com/mrkkrp/req/blob/0c88e2743ba26b1464b60f380e79aaeefb408807/Network/HTTP/Req.hs#L879-L881